### PR TITLE
httpcaddyfile: fix incorrect error message on duplicate matchers

### LINKED ISSUE #7774: Incorrect error message on duplicate request matchers
### Issue Details

Affecting both v2.11.2 and v2.11.3, named matchers (likely, I have no idea if it can be confirmed) no longer get registered asap, causing the syntax below to fail.

```Caddyfile
(preshrankBlob) {
	@{args[0]}_{args[1]} {
		header Accept-Encoding *{args[2]}*
		header !Connection
		path *.{args[0]}
		file {path}.{args[1]}
	}
	handle @{args[0]}_{args[1]} {
		rewrite {path}.{args[1]}
		header Content-Encoding {args[2]}
		header Content-Type {args[3]}
		header X-Served-Path {path}.{args[1]}
		header Cache-Control "min-fresh=20, max-age={args[4]}, no-transform, proxy-revalidate"
		header Vary "Accept-Encoding, Authorization"
	}
}
(preshrankMatrix) {
	import preshrankBlob {args[0]} br br {args[1]} {args[2]}
	import preshrankBlob {args[0]} zst zstd {args[1]} {args[2]}
	import preshrankBlob {args[0]} gz gzip {args[1]} {args[2]}
}

example.com {
	import preshrankMatrix html text/html 120
}
```
```
Error: getting matcher module '@html_br': module not registered: http.matchers.@html_br
```

### Assistance Disclosure

AI not used

### If AI was used, describe the extent to which it was used.

_No response_

### PR DESCRIPTION
**Pull request by Brunotlps**
Fixes #7774

## Problem

When duplicate named matchers exist in a server block, `DispenseDirective`
(parse.go) collects tokens from all segments sharing the same directive
name into a single Dispenser. Inside `parseMatcherDefinitions`, after the
first matcher block is consumed, the token loop picks up the
second `@name` token where a matcher module name (`header`, `path`, etc.)
is expected. This token is passed to `makeMatcher`, which attempts to
resolve `http.matchers.@name`, producing the confusing error:

    module not registered: http.matchers.@name

instead of the correct:

    matcher is defined more than once: @name

## Root Cause

The root cause is in the extraction loop (httptype.go:107-117), which calls
`DispenseDirective(dir)` for each `@`-prefixed segment without tracking
which matcher names have already been processed. This causes duplicate
segments to be concatenated into a single token stream, where the second
definition name is misinterpreted as a module name.

A deeper fix could involve tracking processed matcher names in that loop
to skip duplicates entirely, but that would be a larger change with more
risk of side effects. I'd be happy to explore that in a follow-up if
the maintainers think it's worthwhile.

## Fix

Added a defensive check in `parseMatcherDefinitions` to detect when a
token starting with the matcher prefix `@` appears where a matcher module
name is expected, and return the correct duplicate error message.

## Tests

- Added test case verifying duplicate matchers produce the correct error
  message ("is defined more than once")
- Added regression test with snippets importing distinct named matchers

## AI Disclosure

AI (Claude) was used as a mentoring and learning tool to understand the
codebase architecture and guide the debugging process. All code was
written and fully comprehended by me. I can explain every line.

### CODE REVIEW

File: caddyconfig/httpcaddyfile/httptype.go [hunk 1]
```diff
@@ -1611,6 +1611,9 @@ func parseMatcherDefinitions(d *caddyfile.Dispenser, matchers map[string]caddy.M
 	tokensByMatcherName := make(map[string][]caddyfile.Token)
 	for nesting := d.Nesting(); d.NextArg() || d.NextBlock(nesting); {
 		matcherName := d.Val()
+		if strings.HasPrefix(matcherName, matcherPrefix) {
+			return fmt.Errorf("matcher is defined more than once: %s", definitionName)
+		}
```
Comment (steadytao) [inline caddyconfig/httpcaddyfile/httptype.go]: I think this is the wrong layer. This treats any matcher-name-shaped token here as a duplicate even when the config is just malformed.

For example, `@test { @other { path /x } }` would now report `matcher is defined more than once: @test` but there is no duplicate. I think we should detect duplicates before `DispenseDirective(dir)` coalesces same-name segments or parse each matcher segment directly so the existing duplicate check handles it.

File: caddyconfig/httpcaddyfile/httptype_test.go [hunk 2]
```diff
@@ -53,6 +55,34 @@ func TestMatcherSyntax(t *testing.T) {
 			`,
 			expectError: false,
 		},
+		{
+			input: `http://localhost {
```
Comment (steadytao) [inline caddyconfig/httpcaddyfile/httptype_test.go]: `git diff --check` fails here due to trailing whitespace after `{`

**Review by steadytao — CHANGES_REQUESTED** [review]

File: caddyconfig/httpcaddyfile/httptype.go (hunk 1 above)
Comment (Brunotlps) [reply caddyconfig/httpcaddyfile/httptype.go]: Got it, thanks!

File: caddyconfig/httpcaddyfile/httptype_test.go (hunk 2 above)
Comment (Brunotlps) [reply caddyconfig/httpcaddyfile/httptype_test.go]: Fixed, thanks.

File: caddyconfig/httpcaddyfile/httptype.go (hunk 1 above)
Comment (Brunotlps) [reply caddyconfig/httpcaddyfile/httptype.go]: I've updated the approach. The fix now parses each matcher segment individually with NewDispenser(segment) instead of DispenseDirective(dir). This way the existing duplicate check at the top of parseMatcherDefinitions handles it naturally on the second pass.

**Review by steadytao — APPROVED** [review]
Looks good to me now. This keeps the fix at the matcher extraction layer and lets the existing duplicate definition check handle the error naturally.

Verified the repeated-import shape from #7774 now reports `matcher is defined more than once: @html_br`.

Nice to see some more human interaction in our PRs 😊

**Comment by Brunotlps** [comment]
Thank you for the clear feedback.  I enjoyed working on this and really would like to keep contributing.  If there are other issues that could use a hand, I'm happy to pick them up. Thanks again for the guidance!
